pkg/kudoctl/cmd/package_list_plans.go

```diff
@@ -0,0 +1,129 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/xlab/treeprint"
+
+	"github.com/kudobuilder/kudo/pkg/engine/task"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/env"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
+)
+
+type packageListPlansCmd struct {
+	fs                 afero.Fs
+	out                io.Writer
+	pathOrName         string
+	RepoName           string
+	PackageVersion     string
+	WithTasksResources bool
+}
+
+const (
+	packageListPlansExample = `# show plans from local-folder (where local-folder is a folder in the current directory)
+  kubectl kudo package list plans local-folder
+
+  # show plans from zookeeper (where zookeeper is name of package in KUDO repository)
+  kubectl kudo package list plans zookeeper`
+)
+
+func newPackageListPlansCmd(fs afero.Fs, out io.Writer) *cobra.Command {
+	list := &packageListPlansCmd{fs: fs, out: out}
+
+	cmd := &cobra.Command{
+		Use:     "plans [operator]",
+		Short:   "List operator plans",
+		Example: packageListPlansExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := validateOperatorArg(args); err != nil {
+				return err
+			}
+			list.pathOrName = args[0]
+			return list.run(&Settings)
+		},
+	}
+
+	f := cmd.Flags()
+	f.StringVar(&list.RepoName, "repo", "", "Name of repository configuration to use. (default defined by context)")
+	f.StringVar(&list.PackageVersion, "version", "", "A specific package version on the official GitHub repo. (default to the most recent)")
+	f.BoolVarP(&list.WithTasksResources, "with-tasks", "t", false, "Display task resources with plans")
+
+	return cmd
+}
+
+func (c *packageListPlansCmd) run(settings *env.Settings) error {
+	pf, err := packageDiscovery(c.fs, settings, c.RepoName, c.pathOrName, c.PackageVersion)
+	if err != nil {
+		return err
+	}
+
+	return displayPlanTable(pf.Files, c.WithTasksResources, c.out)
+}
+
+func displayPlanTable(pf *packages.Files, withTasks bool, out io.Writer) error {
+	tree := treeprint.New()
+	planNames := sortedPlanNames(pf)
+	tree.SetValue("plans")
+	for _, name := range planNames {
+		plan := pf.Operator.Plans[name]
+		pNode := tree.AddBranch(fmt.Sprintf("%s (%s)", name, plan.Strategy))
+
+		for _, phase := range plan.Phases {
+			phNode := pNode.AddMetaBranch("phase", fmt.Sprintf("%s (%s)", phase.Name, phase.Strategy))
+			for _, step := range phase.Steps {
+				sNode := phNode.AddMetaBranch("step", step.Name)
+				for _, taskName := range step.Tasks {
+					if withTasks {
+						addTaskNodeWithResources(sNode, taskName, pf)
+					} else {
+						sNode.AddNode(taskName)
+					}
+				}
+			}
+		}
+	}
+
+	var err error
+	if len(pf.Operator.Plans) == 0 {
+		_, err = fmt.Fprintf(out, "no plans found\n")
+	} else {
+		_, err = fmt.Fprintln(out, tree.String())
+	}
+
+	return err
+}
+
+func sortedPlanNames(pf *packages.Files) []string {
+	planNames := make([]string, 0, len(pf.Operator.Plans))
```

Comment:
with go-funk merged, the next few lines could be `planNames := funk.Keys(pf.Operator.Plans)`. the downside would be the resulting type: `var planNames interface{}` so you'll have to case when sorting so choose your poison 🤷‍♂

Reply:
we did and don't have `funk` in the code base yet... but great suggestion

Reply:
we do now 😉 you just to get the latest `master`